### PR TITLE
Add PostgreSQL support in test case pre_deploy_sn

### PR DESCRIPTION
**Pull request by neo954**
Fix issue #2346

### CODE REVIEW

**Comment by tingtli** [comment]
@neo954  Please update linux.template.conf to give example of XCAT_DATABASE. Thanks!

**Comment by zet809** [comment]
hi, @hu-weihua , pls help to review this test case, thx a lot!

File: xCAT-test/autotest/testcase/installation/pre_deploy_sn [hunk 1]
```diff
@@ -16,11 +16,24 @@ sub runcmd {
...
-$cmd = "XCATMYSQLADMIN_PW=12345 XCATMYSQLROOT_PW=12345 /opt/xcat/bin/mysqlsetup -i -V";
-runcmd("$cmd");
-
-$cmd = "echo \"GRANT ALL on xcatdb.* TO xcatadmin@\'%\'  IDENTIFIED BY \'12345\'\;\" | mysql -u root -p12345";
-runcmd("$cmd");
+if ($ENV{XCAT_DATABASE} eq "PostgreSQL") {
+  $cmd = "XCATPGPW=12345 /opt/xcat/bin/pgsqlsetup -i -V";
+  runcmd("$cmd");
+
+  open(my $fh, '>>', '/var/lib/pgsql/data/pg_hba.conf') or die;
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/installation/pre_deploy_sn]: I think there is code in ``runcmd`` to write log. If we just ``die`` here, we can't know what is happened when we debug. Could you find some way to record this error to log file?

**Comment by neo954** [comment]
@hu-weihua, The patch was updated based on your comment. Please review it when you have time. Thanks a lot.

File: xCAT-test/autotest/testcase/installation/pre_deploy_sn [hunk 2]
```diff
@@ -16,11 +16,24 @@ sub runcmd {
...
-$cmd = "XCATMYSQLADMIN_PW=12345 XCATMYSQLROOT_PW=12345 /opt/xcat/bin/mysqlsetup -i -V";
-runcmd("$cmd");
-
-$cmd = "echo \"GRANT ALL on xcatdb.* TO xcatadmin@\'%\'  IDENTIFIED BY \'12345\'\;\" | mysql -u root -p12345";
-runcmd("$cmd");
+if ($ENV{XCAT_DATABASE} eq "PostgreSQL") {
+  $cmd = "XCATPGPW=12345 /opt/xcat/bin/pgsqlsetup -i -V";
+  runcmd("$cmd");
+
+  open(my $fh, '>>', '/var/lib/pgsql/data/pg_hba.conf') or die("failed open file /var/lib/pgsql/data/pg_bha.conf");
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/installation/pre_deploy_sn]: we have code in above file line 5

    cmd:XCAT_DATABASE=$$XCAT_DATABASE /opt/xcat/share/xcat/tools/autotest/testcase/installation/pre_deploy_sn __GETNODEATTR($$SN,os)__  __GETNODEATTR($$SN,arch)__
    check:rc==0
 

So what is the return code of ``die``

**Comment by hu-weihua** [comment]
I have reviewed the change and agree to merge, thanks @neo954 .

**Pull request merged**